### PR TITLE
Unit tests for user error handler

### PR DESCRIPTION
**Pull request by srnagar**
Added unit test to cover error handling scenarios for user callback errors.

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClient.java [hunk 1]
```diff
@@ -40,7 +42,7 @@
     private final ClientLogger logger = new ClientLogger(EventProcessorClient.class);
 
     private final String identifier;
-    private final AtomicBoolean started = new AtomicBoolean(false);
+    private final AtomicBoolean isRunning = new AtomicBoolean(false);
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClient.java]: the no arg constructor is false.  `new AtomicBoolean()`

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClient.java [hunk 2]
```diff
@@ -127,12 +132,22 @@ public synchronized void start() {
...
         this.partitionPumpManager.stopAllPartitionPumps();
     }
+
+    /**
+     * Returns {@code true} if the event processor is running. If the event processor is already running, calling
+     * {@link #start()} has no effect.
+     *
+     * @return {@code true} if the event processor is running.
+     */
+    public synchronized boolean isRunning() {
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClient.java]: Does this need to be synchronised? isRunning.get() is atomic

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionBasedLoadBalancer.java [hunk 3]
```diff
@@ -216,7 +220,11 @@ void loadBalance() {
                 logger.info("This event processor owns {} partitions and shouldn't own more",
                     ownerPartitionMap.get(ownerId).size());
                 // renew ownership of already owned partitions
-                checkpointStore.claimOwnership(ownerPartitionMap.get(this.ownerId)).subscribe();
+                checkpointStore.claimOwnership(partitionPumpManager.getPartitionPumps().keySet()
+                    .stream()
+                    .map(partitionId -> createPartitionOwnershipRequest(partitionOwnershipMap, partitionId))
+                    .collect(Collectors.toList()))
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionBasedLoadBalancer.java]: Curious why we are collecting these into a list? it only emits when the flux is complete.

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClient.java (hunk 2 above)
Comment (srnagar) [reply sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClient.java]: Yes, `isRunning` may have been updated but the start or stop operation may not have been fully completed.

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionBasedLoadBalancer.java (hunk 3 above)
Comment (srnagar) [reply sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionBasedLoadBalancer.java]: `checkpointStore.claimOwnership()` method takes a list of partition ownership instances.

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClient.java (hunk 2 above)
Comment (conniey) [reply sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClient.java]: Ahh, I didn't realise that synchronized is across the entire object. 👍

**Pull request merged**